app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -41,16 +54,88 @@ def get_jwt_identity
         @authentication_parameters.jwt_identity =  @jwt_configuration.jwt_identity(@authentication_parameters)
       end
 
+      def validate_user_has_access_to_webservice
+        @validate_role_can_access_webservice.(
+          webservice: webservice,
+            account: account,
+            user_id: @authentication_parameters.jwt_identity,
+            privilege: 'authenticate'
+        )
+      end
+
+      def validate_origin
+        @validate_origin.(
+          account: account,
+          username: username,
```

Comment:
I'm not sure that `@authenticator_input.username` has value.
In our case as far as I understand `username` is `@authentication_parameters.jwt_identity`.